x/nft/bucket.go

```diff
@@ -12,6 +15,65 @@ type Owned interface {
 	OwnerAddress() weave.Address
 }
 
+//TODO: Better name?
+type BucketAccess interface {
+	Get(db weave.ReadOnlyKVStore, key []byte) (orm.Object, error)
+	Save(db weave.KVStore, model orm.Object) error
+}
+
+type BucketDispatcher interface {
+	Register(t string, bucket BucketAccess) error
+	AssertRegistered(types ...fmt.Stringer)
+	Get(t string) (BucketAccess, error)
+}
+
+type bucketDispatcher struct {
+	mutex     sync.RWMutex
+	bucketMap map[string]BucketAccess
+}
+
+var bs BucketDispatcher
```

Comment:
Okay, so there is a singleton in this package.
I think that is fine, but let's think twice about it.

Reply:
yeah, there is. I can't see a better approach for this to be honest, if you have any suggestions - let know